### PR TITLE
DX-1171 MDCB 2.5 release notes

### PR DESCRIPTION
**Pull request by carolwkc**
### For internal users - Please add a Jira DX PR ticket to the subject!
<br>

[DX-1171](https://tyktech.atlassian.net/browse/DX-1171)

---

### Preview Link

<br>

[preview](https://deploy-preview-4216--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5/)

### Description

<br>

#### Screenshots (if appropriate)
<br>

## Checklist

- [x] I have added a **preview link** to the PR description.
- [x] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I **labelled** the PR

___

## **Type**
documentation, enhancement

___

## **Description**
- Added detailed release notes for Tyk MDCB v2.5, focusing on compatibility updates with Tyk Gateway v5.3 and third-party dependencies like Redis and MongoDB.
- Addressed several fixes, including a critical bug fix for per-API access rights from policies to gateways and security vulnerabilities (CVE-2020-26160).
- Updated the documentation menu to include the new MDCB v2.5 release notes.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>version-2.5.md</strong><dd><code>Tyk MDCB v2.5 Release Notes Addition</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md
<li>Added release notes for Tyk MDCB v2.5.<br> <li> Detailed compatibility with Tyk API Definitions from Tyk Gateway v5.3.<br> <li> Provided upgrade instructions and highlighted new support for Redis <br>v7.0.x and MongoDB v7.0.x.<br> <li> Listed fixed issues including a bug related to per-API access rights <br>and security vulnerabilities.

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4216/files#diff-7b39137d58210cd9bc2105b14da6d8327159ba1daa3dedff47964e050e6378a1">+153/-0</a>&nbsp; </td>
</tr>                    
</table></td></tr><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>menu.yaml</strong><dd><code>Menu Update to Include MDCB v2.5 Release Notes</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/data/menu.yaml
- Updated the menu to include a link to the MDCB v2.5 release notes.

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4216/files#diff-a6789c66bfa0660f7fdbe4796002ab4777677bae59a0276689c840100ff0b0b2">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

[DX-1171]: https://tyktech.atlassian.net/browse/DX-1171?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+#### Downloads
+- [Docker image to pull](https://hub.docker.com/layers/tykio/tyk-mdcb-docker/v2.5.0/images/sha256-TBC?context=explore) <<to be updated>>
+
+#### Changelog {#Changelog-v2.5.0}
+
+##### Fixed
+<ul>
+ <li>
+ <details>
+ <summary>Fixed relaying per-API access rights to gateways</summary>
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md]:
```suggestion
 <summary>Fixed relaying per-API access rights to gateways for MongoDB deployments</summary>
```

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+   
+Fixed a bug in MDCB 2.2.x to 2.4.x that does not relay per-API access rights from policies to gateways.
+It affected GraphQL's field-based permissions, query depth, per query depth limits, and disable introspection
+settings. Also it affected usage quota of both HTTP and GraphQL APIs. However, "Set per API limits and quotas"
+is not affected by this bug.
+
+Note: This bug affected MongoDB users only. PostgreSQL users were not affected.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md]:
```suggestion
Note: This bug affected deployments using MongoDB. PostgreSQL deployments were not affected.
```

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+ </details>
+ </li>
+
+  <li>
+ <details>
+ <summary>TT-10651</summary>
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md]: Should this be removed?

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+   
+ </details>
+ </li>
+
+  <li>
+ <details>
+ <summary>Fixed CVE-2020-26160 (NVD)</summary>
+   
+   Migrate MDCB JWT library to golang-jwt v4.5.0 to address CVE-2020-26160 (NVD)
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md]: Add link to CVE https://nvd.nist.gov/vuln/detail/CVE-2020-26160

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+ </ul>
+
+##### Added
+<ul>
+   <li>
+ <details>
+ <summary>Support Redis v7.0.x</summary>
+   
+   MDCB 2.5.0 refactors Redis connection logic by using [storage v1.2.2](https://github.com/TykTechnologies/storage/releases/tag/v1.2.2), 
+   which integrates with [go-redis](https://github.com/redis/go-redis) v9 underneath which added support to
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md]:
```suggestion
   which integrates with [go-redis](https://github.com/redis/go-redis) v9. Subsequently, this fix adds support for 
```

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+Licensed Protected Product
+
+*This page contains all release notes for version 2.5 displayed in reverse chronological order*
+
+## Support Lifetime
+Our minor releases are supported until our next minor comes out.
+
+## 2.5.0 Release Notes
+
+##### Release date TBC <<to be updated>>
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md]: Remember to add release date when available

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+If you are using a 2.4.x version, we advise you to upgrade ASAP to this latest release. If you are on an older version, you should skip 2.4.0 and upgrade directly to this release.
+
+#### Release Highlights
+MDCB 2.5.0 is an update for compatibility for synchronisation with Tyk v5.3 API Definitions. It has been updated to be compatible with Redis v7.x and MongoDB v7.x.
+It also fixes a bug in MDCB 2.2.x to 2.4.x that does not relay per-API access rights from policies to gateways.
+
+Please refer to the [changelog]({{< ref "#Changelog-v2.5.0">}}) below.
+
+#### Downloads
+- [Docker image to pull](https://hub.docker.com/layers/tykio/tyk-mdcb-docker/v2.5.0/images/sha256-TBC?context=explore) <<to be updated>>
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md]: Remember to update docker image when available

**Review by sredxny — APPROVED** [review]
lgtm 🚀

**Comment by dcs3spp** [comment]
@caroltyk to enable DX review could you kindly raise a Jira ticket in the DX project space and include the ticket ID in the PR title and description

@letzya

**Review by dcs3spp — CHANGES_REQUESTED** [review]
PR LGTM have made minor suggestions

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md (hunk 4 above)
Comment (carolwkc) [reply tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md]: added

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md (hunk 3 above)
Comment (carolwkc) [reply tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md]: updated

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md [hunk 8]
```diff
@@ -101,6 +115,14 @@ Note: This bug affected deployments using MongoDB. PostgreSQL deployments were n
 
 ##### Updated
 <ul>
+ 
+ <li>
+ <details>
+ <summary>Update for compatibility with API definitions for Tyk v5.3</summary>
+
+MDCB supports Tyk API definitions up to Tyk Gateway v5.3.0. Please use this version with Tyk Gateway v5.3.0+.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md]:
```suggestion
MDCB 2.4.x supports Tyk API definitions up to Tyk Gateway v5.3.0. Please use MDCB 2.5.x with Tyk Gateway v5.3.0+.
```

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md [hunk 9]
```diff
@@ -36,8 +36,18 @@ There are no deprecations in this release.
...
 
 #### Release Highlights
-MDCB 2.5.0 is an update for compatibility for synchronisation with Tyk v5.3 API Definitions. It has been updated to be compatible with Redis v7.x and MongoDB v7.x.
-It also fixes a bug in MDCB 2.2.x to 2.4.x that does not relay per-API access rights from policies to gateways.
+
+##### Tyk v5.3 Compatibility
+MDCB 2.5.0 is an update for compatibility for synchronisation with Tyk v5.3 API Definitions.
+
+##### Redis v7.x Compatibility
+We have upgraded redis driver [go-redis](https://github.com/redis/go-redis) to v9. Subsequently, Tyk 5.3 is compatible with Redis v7.x.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md]:
```suggestion
We have upgraded Redis driver [go-redis](https://github.com/redis/go-redis) to v9. Subsequently, Tyk 5.3 is compatible with Redis v7.x.
```

**Review by dcs3spp — CHANGES_REQUESTED** [review]
Thanks @caroltyk I've made minor suggestions and then ready for merge and release. Should this appear in 5.3+ docs only?

**Comment by carolwkc** [comment]
Yes, it can be merged to 5.3 only.

Waiting for release date confirmation.

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+Licensed Protected Product
+
+*This page contains all release notes for version 2.5 displayed in reverse chronological order*
+
+## Support Lifetime
+Our minor releases are supported until our next minor comes out.
+
+## 2.5.0 Release Notes
+
+##### Release date TBC <<to be updated>>
```
Comment (letzya) [inline tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md]:
```suggestion
##### Release date  5 Apr 2024
```

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+##### MongoDB v7.0.x Compatibility
+We have upgraded mongo-go driver to [mongo-go v1.13.1](https://github.com/mongodb/mongo-go-driver/releases/tag/v1.13.1). It allows us to benefit from the bug fixes and enhancements released by MongoDB. We have also tested that both Tyk 5.0.x+ and Tyk 5.3 are compatible with MongoDB v7.0.x.
+
+##### Security Fixes
+We have fixed a security issue affecting MDCB v2.2.0 to v2.4.x, where certain per-API access rights from policies are not properly relayed to edge gateways. We strongly recommend upgrading to MDCB version 2.5.0 to ensure the proper enforcement of per-API access rights across all gateways in your deployment.
+
+Please refer to the [changelog]({{< ref "#Changelog-v2.5.0">}}) below.
+
+#### Downloads
+- [Docker image to pull](https://hub.docker.com/layers/tykio/tyk-mdcb-docker/v2.5.0/images/sha256-TBC?context=explore) <<to be updated>>
```
Comment (letzya) [inline tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md]:
```suggestion
- [Docker image v2.5.0](https://hub.docker.com/r/tykio/tyk-mdcb-docker/tags?page=&page_size=&ordering=&name=v2.5.0) - ```bash
docker pull tykio/tyk-mdcb-docker:v2.5.0``` 
```

**Review by letzya — APPROVED** [review]

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md [hunk 12]
```diff
@@ -0,0 +1,168 @@
...
+Licensed Protected Product
+
+*This page contains all release notes for version 2.5 displayed in reverse chronological order*
+
+## Support Lifetime
+Our minor releases are supported until our next minor comes out.
+
+## 2.5.0 Release Notes
+
+##### Release date  5 Apr 2024
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.5.md]:
```suggestion
##### Release date 5 Apr 2024
```

**Review by dcs3spp — APPROVED** [review]
Thanks @letzya PR LGTM

**Pull request merged**